src/HIE/Bios/Types.hs

```diff
@@ -48,14 +83,32 @@ data ActionName a
   | Other a
   deriving (Show, Eq, Ord, Functor)
 
-data Log =
-  LogAny String
+data Log 
+  = LogAny String
   | LogProcessOutput String
+  | LogCreateProcessRun CreateProcess
+  | LogProcessRun FilePath [FilePath]
   deriving Show
 
 instance Pretty Log where
   pretty (LogAny s) = pretty s
   pretty (LogProcessOutput s) = pretty s
+  pretty (LogProcessRun fp args) = pretty fp <+> pretty (unwords args)
+  pretty (LogCreateProcessRun cp) =
+    vcat $ 
+      [ case cmdspec cp of
+          ShellCommand sh -> pretty sh
+          RawCommand cmd args -> pretty cmd <+> pretty (unwords args)
+      ]
+      <>
+      if null envText
```

Comment:
could put the `indent 2` outside the `if` if you want

Reply:
:scream: thanks!